tests/unit/profiles-management/pmr/test_classes.py

```diff
@@ -0,0 +1,93 @@
+import pytest
+from pydantic import ValidationError
+
+from profiles_management.pmr.classes import (
+    Owner,
+    Profile,
+    ProfilesManagementRepresentation,
+    ResourceQuotaSpecModel,
+    UserKind,
+)
+
+
+@pytest.mark.parametrize(
+    "quota",
+    [
+        {"kimchi": 1},  # arbitrary fields
+        {"scopes": 1},  # incorrect type
+        {"hard": "1000"},  # incorrect type, no keys
+        {"kimchi": 1, "hard": {"cpu": "1000"}},  # arbitrary field plus correct field
+    ],
+)
+def test_invalid_resource_quota(quota):
+    with pytest.raises(ValidationError):
+        Profile(
+            name="test",
+            resources=ResourceQuotaSpecModel.model_validate(quota),
+            owner=Owner(name="kimchi", kind=UserKind.USER),
+        )
+
+
+def test_valid_resource_quota():
+    profile = Profile(
+        name="test",
+        contributors=[],
+        resources=ResourceQuotaSpecModel.model_validate(
+            {
+                "hard": {"cpu": "1000"},
+                "scopes": ["test"],
+            }
+        ),
+        owner=Owner(name="kimchi", kind=UserKind.USER),
+    )
+
+    assert profile.resources is not None
+    assert profile.resources.hard is not None
+    assert profile.resources.hard["cpu"] == "1000"
+
+
+def test_profiles_in_pmr():
```

Comment:
This test case is a candidate for `parametrize` as well, you'd have the profiles and the expected bool as parameters. I'll let you chose, but it will definitely be nicer.

Reply:
I think it's already good enough.